lib/prawn/icon/interface.rb

```diff
@@ -225,7 +225,17 @@ def format_hash
 
     def render
       @pdf.font(@data.path) do
-        @pdf.text @unicode, @options
+        opts = @options.dup
+
+        size_mode = opts.delete(:size_mode) || :font_size
+        if size_mode == :icon_size
```

Comment:
It's hard to grasp what the `:icon_size` option really means (and how it differs from the default of `:font_size`).

In my opinion, size modes of `:default` and `:scaled` are easier to digest as a newcomer to the library. 

Alternatively we could utilize a boolean option such as `autoscale: true`.

Reply:
I was going for "how am I going to interpret `:size`". `:font_size` is what you currently get by simply passing `:size` on to Prawn. `:icon_size` (or maybe better `:icon_height`) would then mean that `:size` is interpreted as the desired size/height for the icon.

I'm wondering what would most clearly communicate intent at the call site:

```
icon(:size => 24, :size_mode => :default)
icon(:size => 24, :size_mode => :scaled)
icon(:size => 24, :size_is => :font_size)
icon(:size => 24, :size_is => :icon_height)
icon(:size => 24, :autoscale => true)
```

`:size_mode => :scaled` and `:autoscale => true` feel a bit too vague for me.

Happy to go with whatever you prefer though.

Reply:
> :size_mode => :scaled and :autoscale => true feel a bit too vague for me.


True - though I really like your suggestion of `:font_size` and `:icon_height`. Let's go with that 👍 

i.e.

```ruby
icon("far-circle", size: 24, size_mode: :icon_height)
icon("far-circle", size: 24, size_mode: :font_size)
```

Reply:
Done.